nailgun/entity_mixins.py

```diff
@@ -784,6 +784,9 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
             ignore = set()
 
         for field_name, field in entity.get_fields().items():
+            if field_name == 'environment':
+                if field.entity.__name__ == 'Environment':
```

Comment:
Why is this double condition needed? Is this to separate the Lifecycle environment from Puppet Environment?

if not stick the lower condition only.

Reply:
@jyejare yeah, I don't like it either. But when I tried to use both of them together it failed with error saying `field` has no object `entity`, and that's why this workaround.

Reply:
@jameerpathan111 Cool! I am almost fell settled with this app! But,

Historically, we never set / unset things like this for components outside `entities.py` and hence we need to decide if this is the right place to globally ignore the depreciated fields/components from entity_mixins or any other module as this may look scattered. So, let's talk on JFT call if we should:
1. Go with this way to ignore depreciated fields globally / Base class. (Which I like and prefer)
2. Or, ignore the field from the entity classes read the definition, which probably dont need but if component owners(those who needs `environment` field) can take the decision.

Finally, @jameerpathan111 if we wanted to go with approach 1 modify the code as:
```
# Class level list of ignore / depreciated fields
ignore_fields = ['environment', 'etc']
```
and finally:
```suggestion
            if field_name in ignore_fields and field.entity.__name__.lower() in ignore_fields:
```

Reply:
@jyejare entity can be referred using more than one name. For ex. LifecycleEnvironment is referred using `environment`, `library`, `prior` and `lifecycle_environment`. How about doing it like this:

```
# Class level list of ignore / depreciated fields
    ignore_fields = ['environment']  
    ignore_entity = ['Environment']
```

```
            if field_name in self.ignore_fields and field.entity.__name__ in self.ignore_entity:
                continue
```
And we can ignore the double if condition, it's working fine without it.

Reply:
@jyejare I have updated the pr.